src/Microsoft.Azure.WebJobs.Host.Storage/StorageServiceCollectionExtensions.cs

```diff
@@ -2,90 +2,59 @@
 
 namespace Microsoft.Extensions.Hosting
 {
     public static class StorageServiceCollectionExtensions
     {
-        [Obsolete("Dashboard is being deprecated. Use AppInsights.")]
-        public static IServiceCollection AddDashboardLogging(this IServiceCollection services)
-        {
-            services.TryAddSingleton<LoggerProviderFactory>();
-
-            services.TryAddSingleton<IFunctionOutputLoggerProvider>(p => p.GetRequiredService<LoggerProviderFactory>().GetLoggerProvider<IFunctionOutputLoggerProvider>());
-            services.TryAddSingleton<IFunctionOutputLogger>(p => p.GetRequiredService<IFunctionOutputLoggerProvider>().GetAsync(CancellationToken.None).GetAwaiter().GetResult());
-
-            services.TryAddSingleton<IFunctionInstanceLoggerProvider>(p => p.GetRequiredService<LoggerProviderFactory>().GetLoggerProvider<IFunctionInstanceLoggerProvider>());
-            services.TryAddSingleton<IFunctionInstanceLogger>(p => p.GetRequiredService<IFunctionInstanceLoggerProvider>().GetAsync(CancellationToken.None).GetAwaiter().GetResult());
-
-            services.TryAddSingleton<IHostInstanceLoggerProvider>(p => p.GetRequiredService<LoggerProviderFactory>().GetLoggerProvider<IHostInstanceLoggerProvider>());
-            services.TryAddSingleton<IHostInstanceLogger>(p => p.GetRequiredService<IHostInstanceLoggerProvider>().GetAsync(CancellationToken.None).GetAwaiter().GetResult());
-
-            return services;
-        }
-
         public static void AddAzureStorageCoreServices(this IServiceCollection services)
         {
             // Replace existing runtime services with storage-backed implementations.
             // Add runtime services that depend on storage.
             services.AddSingleton<IDistributedLockManager>(provider => Create(provider));
 
-            // Used specifically for the CloudBlobContainerDistributedLockManager implementation
-            services.TryAddSingleton<DistributedLockManagerContainerProvider>();
-
-            services.TryAddEnumerable(ServiceDescriptor.Transient<IConfigureOptions<StorageAccountOptions>, StorageAccountOptionsSetup>());
             services.TryAddEnumerable(ServiceDescriptor.Transient<IConfigureOptions<JobHostInternalStorageOptions>, CoreWebJobsOptionsSetup<JobHostInternalStorageOptions>>());
 
             services.TryAddSingleton<IDelegatingHandlerProvider, DefaultDelegatingHandlerProvider>();
 
+            // May need to rename this to HostBlobServiceClientProvider
+            services.TryAddSingleton<BlobServiceClientProvider>();
+            services.AddAzureClientsCore();
+            services.TryAddSingleton<IAzureStorageProvider, AzureStorageProvider>();
+
             services.AddSingleton<IConcurrencyStatusRepository, BlobStorageConcurrencyStatusRepository>();
         }
 
         // This is only called if the host didn't already provide an implementation
+        // TODO: Should this attempt to create a client or just check for connection setting?
```

Comment:
Well, you're replicating the existing behavior. What are you suggesting instead?

Reply:
I remember this point being brought up. There's 3 ways to validate a connection as successful. 
1. Connection value is present (i.e. is the setting defined in app settings?)
2. Connection value can be used to create a client (i.e. does the connection value parsing work to create the client?)
3. Connection value can be used to talk to storage (i.e. make a network call and success means connection is valid). 

The existing behavior (and kept with the changes here) uses approach #2.